src/python/pants/engine/unions.py

```diff
@@ -148,7 +148,7 @@ def is_member(self, union_type: type, putative_member: type) -> bool:
         members = self.union_rules.get(union_type)
         if members is None:
             raise TypeError(f"Not a registered union type: {union_type}")
-        return type(putative_member) in members
```

Comment:
This was absolutely never correct: `putative_member` is already a `type`. It only worked because we passed the wrong thing in and somehow the typechecker didn't notice.